server/tserver/src/main/java/org/apache/accumulo/tserver/tablet/Compactor.java

```diff
@@ -388,9 +389,10 @@ else if (env.getIteratorScope() == IteratorScope.minc)
           }
           throw new CompactionCanceledException();
         }
-
+      } catch (Exception e) {
+        throw new IOException(e);
       } finally {
-        CompactionStats lgMajcStats = new CompactionStats(citr.getCount(), entriesCompacted);
+        CompactionStats lgMajcStats = new CompactionStats(statsCount, entriesCompacted);
```

Comment:
Careful about suppressed exceptions here, especially when using multiple resources in a single try-with-resources, and mixing try-with-resources with a finally block. Any exception thrown in the finally block will suppress try-with-resources, and exception from closing one resource may result in others being suppressed.

How do we want to handle suppressed exceptions? Do we assume they've already been logged earlier, or do we try to handle them here? Wrapping `Exception e` with `IOException` may also make it more difficult for the caller to track the suppressed exceptions attached to `e`, which is now in the `getCause()` of the `IOException`.

Also, statsCount might be stale (or uninitialized) in the finally block if an exception occurs in the try block.